pyomo/contrib/solver/tests/solvers/test_solvers.py

```diff
@@ -0,0 +1,1623 @@
+        self.assertAlmostEqual(m.y.value, 3)
+        m.x.value = 0
+        res = opt.solve(m)
+        self.assertAlmostEqual(m.x.value, 0)
+        self.assertAlmostEqual(m.y.value, 2)
+
+    @parameterized.expand(input=_load_tests(all_solvers))
+    def test_fixed_vars_3(
+        self, name: str, opt_class: Type[SolverBase], use_presolve: bool
+    ):
+        opt: SolverBase = opt_class()
+        if opt.is_persistent():
+            opt.config.auto_updates.treat_fixed_vars_as_params = True
+        if not opt.available():
+            raise unittest.SkipTest(f'Solver {opt.name} not available.')
+        if any(name.startswith(i) for i in nl_solvers_set):
+            if use_presolve:
+                opt.config.writer_config.linear_presolve = True
+            else:
+                opt.config.writer_config.linear_presolve = False
+        m = pe.ConcreteModel()
+        m.x = pe.Var()
+        m.y = pe.Var()
+        m.obj = pe.Objective(expr=m.x + m.y)
+        m.c1 = pe.Constraint(expr=m.x == 2 / m.y)
+        m.y.fix(1)
+        res = opt.solve(m)
+        self.assertAlmostEqual(res.incumbent_objective, 3)
+        self.assertAlmostEqual(m.x.value, 2)
+
+    @parameterized.expand(input=_load_tests(nlp_solvers))
+    def test_fixed_vars_4(
+        self, name: str, opt_class: Type[SolverBase], use_presolve: bool
+    ):
+        opt: SolverBase = opt_class()
+        if opt.is_persistent():
+            opt.config.auto_updates.treat_fixed_vars_as_params = True
+        if not opt.available():
+            raise unittest.SkipTest(f'Solver {opt.name} not available.')
+        if any(name.startswith(i) for i in nl_solvers_set):
+            if use_presolve:
+                opt.config.writer_config.linear_presolve = True
+            else:
+                opt.config.writer_config.linear_presolve = False
+        m = pe.ConcreteModel()
+        m.x = pe.Var()
+        m.y = pe.Var()
+        m.obj = pe.Objective(expr=m.x**2 + m.y**2)
+        m.c1 = pe.Constraint(expr=m.x == 2 / m.y)
+        m.y.fix(1)
+        res = opt.solve(m)
+        self.assertAlmostEqual(m.x.value, 2)
+        m.y.unfix()
+        res = opt.solve(m)
+        self.assertAlmostEqual(m.x.value, 2**0.5)
+        self.assertAlmostEqual(m.y.value, 2**0.5)
+
+    @parameterized.expand(input=_load_tests(all_solvers))
+    def test_mutable_param_with_range(
+        self, name: str, opt_class: Type[SolverBase], use_presolve: bool
+    ):
+        opt: SolverBase = opt_class()
+        if not opt.available():
+            raise unittest.SkipTest(f'Solver {opt.name} not available.')
+        if any(name.startswith(i) for i in nl_solvers_set):
+            if use_presolve:
+                opt.config.writer_config.linear_presolve = True
+            else:
+                opt.config.writer_config.linear_presolve = False
+        try:
+            import numpy as np
+        except:
+            raise unittest.SkipTest('numpy is not available')
```

Comment:
Do you need this check? Isn't the test class is skipped entirely when numpy is not available?